examples/reference/custom_components/PreactComponent.md

```diff
@@ -0,0 +1,401 @@
+```
+
+See [import map](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script/type/importmap) for more info.
+
+## External Files
+
+You can load JavaScript and CSS from files by providing the paths to these files.
+
+Create the file **counter_button.py**.
+
+```python
+from pathlib import Path
+
+import param
+import panel as pn
+
+pn.extension()
+
+class CounterButton(pn.PreactComponent):
+
+    value = param.Integer()
+
+    _esm = Path("counter_button.js")
+    _stylesheets = [Path("counter_button.css")]
+
+CounterButton().servable()
+```
+
+Now create the file **counter_button.js**.
+
+```javascript
+export function render({ html, data }) {
+    return html`
+        <button onClick=${() => data.value += 1}>
+            count is ${data.value}
+        </button>
+    `;
+}
+```
+
+Now create the file **counter_button.css**.
+
+```css
+button {
+    background: #0072B5;
+    color: white;
+    border: none;
+    padding: 10px;
+    border-radius: 4px;
+}
+button:hover {
+    background: #4099da;
+}
+```
+
+Serve the app with `panel serve counter_button.py --autoreload`.
+
+You can now edit the JavaScript or CSS file, and the changes will be automatically reloaded.
+
+- Try changing the `innerHTML` from `count is ${data.value}` to `COUNT IS ${data.value}` and observe the update.
+- Try changing the background color from `#0072B5` to `#008080`.
+
+## Displaying A Single Panel Component
+
+You can display Panel components by defining `ClassSelector` parameters with the `class_` set to subtype of `_Viewable` or tuple of subtypes of `_Viewable`s.
+
+Lets start with the simplest example
+
+```python
+import param
+import panel as pn
+
+class Example(pn.PreactComponent):
+
+    child = param.ClassSelector(class_=pn.viewable.Viewable)
+
+    _esm = """
+    export function render({ html, children }) {
+      return html`
+        <button ref=${ref => ref && ref.appendChild(children.child)}>
+        </button>
+    `}
+    """
+
+Example(child=pn.panel("A **Markdown** pane!")).servable()
+```
+
+If you want to allow a certain type of Panel components only you can specify the specific type in the `_class` argument.
+
+```python
+import param
+import panel as pn
+
+class Example(pn.PreactComponent):
+
+    child = param.ClassSelector(class_=pn.viewable.Markdown)
+
+    _esm = """
+    export function render({ html, children }) {
+      return html`
+        <button ref=${ref => ref && ref.appendChild(children.child)}>
+        </button>
+    `}
+    """
+
+Example(child=pn.panel("A **Markdown** pane!")).servable()
+```
+
+The `class_` argument also supports a tuple of types
+
+DOES NOT WORK YET! PLEASE FIX.
+
+```python
+import param
+import panel as pn
+
+class Example(pn.PreactComponent):
+
+    child = param.ClassSelector(class_=(pn.pane.Markdown, pn.pane.HTML))
+
+    _esm = """
+    export function render({ html, children }) {
+      return html`
+        <button ref=${ref => ref && ref.appendChild(children.child)}>
```

Comment:
Preact examples with children are more complicated than React examples because I can't just insert `${children.child}`. Instead I have to use the `ref` function.

Could it be made easier?

Reply:
Done, it's now identical to React.